survey_runner.tf

```diff
@@ -0,0 +1,190 @@
+    AWS_DEFAULT_REGION                   = "${var.aws_default_region}"
+    EQ_UA_ID                             = "${var.google_analytics_code}"
+    EQ_SECRETS_FILE                      = "${var.secrets_file_name}"
+    LOG_GROUP                            = "${aws_cloudwatch_log_group.survey_runner.name}"
+    CONTAINER_REGISTRY                   = "${var.docker_registry}"
+    CONTAINER_TAG                        = "${var.survey_runner_tag}"
+  }
+}
+
+resource "aws_ecs_task_definition" "survey_runner" {
+  family                = "${var.env}-survey-runner"
+  container_definitions = "${data.template_file.survey_runner.rendered}"
+  task_role_arn         = "${aws_iam_role.survey_runner_task.arn}"
+}
+
+resource "aws_ecs_service" "survey_runner" {
+  depends_on = [
+    "aws_alb_target_group.survey_runner",
+    "aws_alb_listener_rule.survey_runner",
+  ]
+
+  name            = "${var.env}-survey-runner"
+  cluster         = "${data.aws_ecs_cluster.ecs-cluster.id}"
+  task_definition = "${aws_ecs_task_definition.survey_runner.arn}"
+  desired_count   = "${var.survey_runner_min_tasks}"
+  iam_role        = "${aws_iam_role.survey_runner.arn}"
+
+  placement_strategy {
+    type  = "spread"
+    field = "host"
+  }
+
+  load_balancer {
+    target_group_arn = "${aws_alb_target_group.survey_runner.arn}"
+    container_name   = "survey-runner"
+    container_port   = 5000
+  }
+
+  lifecycle {
+    ignore_changes = ["placement_strategy"]
+  }
+}
+
+resource "aws_iam_role" "survey_runner" {
+  name = "${var.env}_iam_for_survey_runner"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": ["ecs.amazonaws.com", "ecs-tasks.amazonaws.com"]
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+EOF
+}
+
+data "aws_iam_policy_document" "survey_runner" {
+  "statement" = {
+    "effect" = "Allow"
+
+    "actions" = [
+      "elasticloadbalancing:*",
+    ]
+
+    "resources" = [
+      "*",
+    ]
+  }
+}
+
+resource "aws_iam_role_policy" "survey_runner" {
+  name   = "${var.env}_iam_for_survey_runner"
+  role   = "${aws_iam_role.survey_runner.id}"
+  policy = "${data.aws_iam_policy_document.survey_runner.json}"
+}
+
+resource "aws_iam_role" "survey_runner_task" {
+  name = "${var.env}_iam_for_survey_runner_task"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": ["ecs-tasks.amazonaws.com"]
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+EOF
+}
+
+data "aws_iam_policy_document" "survey_runner_task" {
+  "statement" = {
+    "effect" = "Allow"
+
+    "actions" = [
+      "s3:GetObject",
+      "s3:ListObjects",
+      "s3:ListBucket",
+    ]
+
+    "resources" = [
+      "arn:aws:s3:::${var.s3_secrets_bucket}",
+      "arn:aws:s3:::${var.s3_secrets_bucket}/*",
```

Comment:
The bucket will only be used for survey runner secrets so there is no reason this shouldn't be allowed to read everything. and if we split the secrets and keys to multiple files this will be simpler